system_metrics_collector/test/moving_average_statistics/test_moving_average_statistics.cpp

```diff
@@ -0,0 +1,198 @@
+// Copyright 2019 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+
+#include <gtest/gtest.h>
+
+#include <memory>
+#include <thread>
+#include <atomic>
+#include "moving_average_statistics/moving_average.hpp"
+
+
+/**
+ * Test fixture
+ */
+class MovingAverageStatisticsTestFixture : public ::testing::Test
+{
+public:
+  void SetUp() override
+  {
+    moving_average_statistics = std::make_shared<MovingAverageStatistics<double>>();
+  }
+
+  void TearDown() override
+  {
+    moving_average_statistics->reset();
+  }
+
+protected:
+  std::shared_ptr<MovingAverageStatistics<double>> moving_average_statistics = nullptr;
+  int expected_count = 0;
+};
+
+// Useful testing constants
+const double TEST_DATA[] = {-3.5, -2.1, -1.1, 0.0, 4.7, 5.0, 6.7, 9.9, 11.0};
+const double EXPECTED_AVG = 3.4;
+const double EXPECTED_MIN = -3.5;
+const double EXPECTED_MAX = 11.0;
+const double EXPECTED_STD = 4.997999599839919955173;
```

Comment:
What are the chances we get flakey tests on different OS b/c of precision?

Reply:
> What are the chances we get flakey tests on different OS b/c of precision?

With the current tests and thresholds I believe low. I verified this by running the tests with 

```bash
'time ./test_moving_average_statistics --gtest_repeat=2000 --gtest_shuffle --gtest_break_on_failure'
```